operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -144,27 +154,87 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		retrigErr = err
 	}
 
+
+
+	keepNsOpen := true; // keepNsOpen defines if we should close the personal namespace based on the last login date
+	                    // must be initialized for later use
+
+	// If the personalNamespace is not created and the lastLogin is blank, then this is the first time
+	// the tenant has logged in; update the lastLogin date to current time
+	if tn.Status.PersonalNamespace.Created==false && tn.Spec.LastLogin=="" {
+
+	  tn.Spec.LastLogin = time.Now().Format(time.RFC3339)
+      klog.Infof("First login of %s: setting lastLogin value of %s in tenant spec.", tn.Name, tn.Spec.LastLogin)
+
+	} else { // Otherwise, this is not the first time the user has logged in
+
+	  // So we check to see if last login was more than TENANT_WORKSPACE_KEEP_ALIVE_SECONDS in the past:
+	  // if so, temporarily delete the namespace
+      
+	  t, err := time.Parse(time.RFC3339, tn.Spec.LastLogin) // Read lastLogin string and converte to datetime
+	  if err != nil {
+		  klog.Errorf("Error in parsing lastLogin date in RFC3339 format of tenant %s -> %s", tn.Name, err)
+		  return ctrl.Result{}, retrigErr
+	  }
+
+      // Calculate time elapsed since lastLogin (now minus lastLogin in seconds)
+	  sPassed := int(time.Now().Sub(t).Seconds());
+
+	  klog.Infof("Last login of tenant %s was %d seconds ago", tn.Name, sPassed)
+
+	  if(sPassed>TENANT_WORKSPACE_KEEP_ALIVE_SECONDS) { // seconds
```

Comment:
Same here.

Reply:
Resolved in commit [43980c7](https://github.com/netgroup-polito/CrownLabs/pull/821/commits/43980c707a44b8d1852c95a87eaebd24843c859e)